crypto/kzg/src/lib.rs

```diff
@@ -51,18 +51,41 @@ impl From<c_kzg::Error> for Error {
 #[derive(Debug)]
 pub struct Kzg {
     trusted_setup: KzgSettings,
-    context: Option<DASContext>,
+    context: DASContext,
 }
 
 impl Kzg {
+    pub fn new_from_trusted_setup_no_precomp(trusted_setup: TrustedSetup) -> Result<Self, Error> {
+        let peerdas_trusted_setup = PeerDASTrustedSetup::from(&trusted_setup);
+
+        let context = DASContext::new(&peerdas_trusted_setup, rust_eth_kzg::UsePrecomp::No);
+
+        Ok(Self {
+            trusted_setup: KzgSettings::load_trusted_setup(
+                &trusted_setup.g1_points(),
+                &trusted_setup.g2_points(),
+            )?,
+            context,
+        })
+    }
+
     /// Load the kzg trusted setup parameters from a vec of G1 and G2 points.
     pub fn new_from_trusted_setup(trusted_setup: TrustedSetup) -> Result<Self, Error> {
+        let peerdas_trusted_setup = PeerDASTrustedSetup::from(&trusted_setup);
```

Comment:
This function is now identical to `new_from_trusted_setup_das_enabled`.
I think maybe what we want is
- if PeerDAS scheduled, load it with DASContext
- Else load with no DASContext OR no precomputation (although I prefer the former as initilisation of `PeerDASTrustedSetup` still require cloning trusted setup bytes.